src/EntityFramework.Storage/Stores/PersistedGrantStore.cs

```diff
@@ -55,31 +55,25 @@ public PersistedGrantStore(IPersistedGrantDbContext context, ILogger<PersistedGr
     public virtual async Task StoreAsync(Duende.IdentityServer.Models.PersistedGrant token)
     {
         using var activity = Tracing.StoreActivitySource.StartActivity("PersistedGrantStore.Store");
-        
-        var existing = (await Context.PersistedGrants.Where(x => x.Key == token.Key)
-                .ToArrayAsync(CancellationTokenProvider.CancellationToken))
-            .SingleOrDefault(x => x.Key == token.Key);
-        if (existing == null)
-        {
-            Logger.LogDebug("{persistedGrantKey} not found in database", token.Key);
 
-            var persistedGrant = token.ToEntity();
-            Context.PersistedGrants.Add(persistedGrant);
-        }
-        else
-        {
-            Logger.LogDebug("{persistedGrantKey} found in database", token.Key);
-
-            token.UpdateEntity(existing);
-        }
+        var persistedGrant = token.ToEntity();
+        Context.PersistedGrants.Add(persistedGrant);
 
         try
         {
             await Context.SaveChangesAsync(CancellationTokenProvider.CancellationToken);
+            Logger.LogDebug("{persistedGrantKey} not found in database", token.Key);
         }
         catch (DbUpdateConcurrencyException ex)
         {
-            Logger.LogWarning("exception updating {persistedGrantKey} persisted grant in database: {error}", token.Key, ex.Message);
+            Logger.LogWarning("exception inserting or updating {persistedGrantKey} persisted grant in database: {error}", token.Key, ex.Message);
+        }
+        catch (DbUpdateException)
+        {
+            Logger.LogDebug("{persistedGrantKey} found in database", token.Key);
+            await Context.PersistedGrants.Where(x => x.Key == key).ExecuteDeleteAsync(CancellationTokenProvider.CancellationToken);
+            Logger.LogDebug("removed {persistedGrantKey} persisted grant from database", key);
+            await Context.SaveChangesAsync(CancellationTokenProvider.CancellationToken);
```

Comment:
This _could_ be done with ExecuteUpdate, but it does not seem to allow for updating the entire entity, but rather each property. Requiring any additional properties in the future to be included in the future -- so not as maintainable.